src/libraries/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceBase.cs

```diff
@@ -72,6 +72,26 @@ public unsafe void RequestAdditionalTime(int milliseconds)
             }
         }
 
+#if NETCOREAPP
+        /// <summary>
+        /// When this method is called from OnStart, OnStop, OnPause or OnContinue,
+        /// the specified wait hint is passed to the
+        /// Service Control Manager to avoid having the service marked as not responding.
+        /// </summary>
+        /// <param name="time"></param>
+        public void RequestAdditionalTime(TimeSpan time) => RequestAdditionalTime(ToTimeoutMilliseconds(time));
+
+        private static int ToTimeoutMilliseconds(TimeSpan timeout)
```

Comment:
In this case, it isn't a timeout, it's requested extra time.

Maybe change all of these methods to "ToIntMilliseconds(TimeSpan time)" "ToIntMicroseconds(TimeSpan time)" etc?